### PR TITLE
feat(prometheus): add basic_auth and response timeout environment var

### PR DESCRIPTION
**Pull request by aurrelhebert**

### CODE REVIEW

File: administrate/metrics/overview.md [hunk 1]
```diff
@@ -558,6 +558,15 @@ If needed, you can override those settings with the two following environment va
...
 - `CC_METRICS_PROMETHEUS_PATH`: Define the path on which the Prometheus endpoint is available
 
+As with Prometheus the exposed host can be the same as the application deployed, you can use a basic authentication to collect the metrics with the two following environment variables:
+
+- `CC_METRICS_PROMETHEUS_USER`: Define the user for the basic auth of the Prometheus endpoint
+- `CC_METRICS_PROMETHEUS_PASSWORD`: Define the password for the basic auth of the Prometheus endpoint
+
+For large custom set of metrics to collect, the default response timeout of the `/metrics` query is 3 seconds. You can update it with the following environment variable:
+
+- `CC_METRICS_PROMETHEUS_RESPONSE_TIMEOUT_SECONDS`: Define the timout in seconds to collect the application metrics. This value **must** be below 60 seconds as data are collected every minutes.
```
Comment (gaelreyrol) [inline administrate/metrics/overview.md:568]:
```suggestion
- `CC_METRICS_PROMETHEUS_RESPONSE_TIMEOUT`: Define the timout in seconds to collect the application metrics. This value **must** be below 60 seconds as data are collected every minutes. 
```

File: reference/reference-environment-variables.md [hunk 2]
```diff
@@ -55,6 +55,9 @@ So you can alter the build&start process for your application.
...
 |[CC_METRICS_PROMETHEUS_PORT]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the port on which the Prometheus endpoint is available | `8080` |
 |[CC_METRICS_PROMETHEUS_PATH]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the path on which the Prometheus endpoint is available | `/metrics` |
+|[CC_METRICS_PROMETHEUS_USER]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the user for the basic auth of the Prometheus endpoint | `None` |
+|[CC_METRICS_PROMETHEUS_PASSWORD]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the password for the basic auth of the Prometheus endpoint| `None` |
+|[CC_METRICS_PROMETHEUS_RESPONSE_TIMEOUT_SECONDS]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the timout in seconds to collect the application metrics. This value **must** be below 60 seconds as data are collected every minutes | `3` |
```
Comment (gaelreyrol) [inline reference/reference-environment-variables.md:60]:
```suggestion
|[CC_METRICS_PROMETHEUS_RESPONSE_TIMEOUT]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the timout in seconds to collect the application metrics. This value **must** be below 60 seconds as data are collected every minutes | `3` |
```

File: reference/reference-environment-variables.md [hunk 3]
```diff
@@ -55,6 +55,9 @@ So you can alter the build&start process for your application.
 |[CC_OVERRIDE_BUILDCACHE]({{< ref "develop/env-variables.md#settings-you-can-define-using-environment-variables" >}}) | Allows to specify paths that will be in the build cache. <br />Only those files / directories will be cached |  |
 |[CC_METRICS_PROMETHEUS_PORT]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the port on which the Prometheus endpoint is available | `8080` |
 |[CC_METRICS_PROMETHEUS_PATH]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the path on which the Prometheus endpoint is available | `/metrics` |
+|[CC_METRICS_PROMETHEUS_USER]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the user for the basic auth of the Prometheus endpoint | `None` |
+|[CC_METRICS_PROMETHEUS_PASSWORD]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the password for the basic auth of the Prometheus endpoint| `None` |
```
Comment (gaelreyrol) [inline reference/reference-environment-variables.md:59]:
```suggestion
|[CC_METRICS_PROMETHEUS_PASSWORD]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the password for the basic auth of the Prometheus endpoint|  |
```

File: reference/reference-environment-variables.md [hunk 4]
```diff
@@ -55,6 +55,9 @@ So you can alter the build&start process for your application.
 |[CC_OVERRIDE_BUILDCACHE]({{< ref "develop/env-variables.md#settings-you-can-define-using-environment-variables" >}}) | Allows to specify paths that will be in the build cache. <br />Only those files / directories will be cached |  |
 |[CC_METRICS_PROMETHEUS_PORT]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the port on which the Prometheus endpoint is available | `8080` |
 |[CC_METRICS_PROMETHEUS_PATH]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the path on which the Prometheus endpoint is available | `/metrics` |
+|[CC_METRICS_PROMETHEUS_USER]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the user for the basic auth of the Prometheus endpoint | `None` |
```
Comment (gaelreyrol) [inline reference/reference-environment-variables.md:58]:
```suggestion
|[CC_METRICS_PROMETHEUS_USER]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the user for the basic auth of the Prometheus endpoint |  |
```

Comment (gaelreyrol) [reply reference/reference-environment-variables.md:58]: For now if the value is not defined by default we leave the field empty.

File: administrate/metrics/overview.md [hunk 5]
```diff
@@ -558,6 +558,15 @@ If needed, you can override those settings with the two following environment va
...
 - `CC_METRICS_PROMETHEUS_PATH`: Define the path on which the Prometheus endpoint is available
 
+As with Prometheus the exposed host can be the same as the application deployed, you can use a basic authentication to collect the metrics with the two following environment variables:
+
+- `CC_METRICS_PROMETHEUS_USER`: Define the user for the basic auth of the Prometheus endpoint
+- `CC_METRICS_PROMETHEUS_PASSWORD`: Define the password for the basic auth of the Prometheus endpoint
+
+For large custom set of metrics to collect, the default response timeout of the `/metrics` query is 3 seconds. You can update it with the following environment variable:
+
+- `CC_METRICS_PROMETHEUS_RESPONSE_TIMEOUT`: Define the timout in seconds to collect the application metrics. This value **must** be below 60 seconds as data are collected every minutes.
```
Comment (BlackYoup) [inline administrate/metrics/overview.md:568]:
```suggestion
- `CC_METRICS_PROMETHEUS_RESPONSE_TIMEOUT`: Define the timeout in seconds to collect the application metrics. This value **must** be below 60 seconds as data are collected every minutes. 
```

File: reference/reference-environment-variables.md [hunk 6]
```diff
@@ -55,6 +55,9 @@ So you can alter the build&start process for your application.
 |[CC_OVERRIDE_BUILDCACHE]({{< ref "develop/env-variables.md#settings-you-can-define-using-environment-variables" >}}) | Allows to specify paths that will be in the build cache. <br />Only those files / directories will be cached |  |
 |[CC_METRICS_PROMETHEUS_PORT]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the port on which the Prometheus endpoint is available | `8080` |
 |[CC_METRICS_PROMETHEUS_PATH]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the path on which the Prometheus endpoint is available | `/metrics` |
+|[CC_METRICS_PROMETHEUS_USER]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the user for the basic auth of the Prometheus endpoint | |
+|[CC_METRICS_PROMETHEUS_PASSWORD]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the password for the basic auth of the Prometheus endpoint| |
```
Comment (BlackYoup) [inline reference/reference-environment-variables.md:59]:
```suggestion
|[CC_METRICS_PROMETHEUS_PASSWORD]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the password for the basic auth of the Prometheus endpoint | |
```

File: reference/reference-environment-variables.md [hunk 7]
```diff
@@ -55,6 +55,9 @@ So you can alter the build&start process for your application.
...
+|[CC_METRICS_PROMETHEUS_RESPONSE_TIMEOUT]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the timout in seconds to collect the application metrics. This value **must** be below 60 seconds as data are collected every minutes | `3` |
```
Comment (BlackYoup) [inline reference/reference-environment-variables.md:60]:
```suggestion
|[CC_METRICS_PROMETHEUS_RESPONSE_TIMEOUT]({{< ref "administrate/metrics/overview.md#publish-your-own-metrics" >}}) | Define the timeout in seconds to collect the application metrics. This value **must** be below 60 seconds as data are collected every minutes | `3` |
```

**Review by BlackYoup — CHANGES_REQUESTED** [review]
Few typos, other than that, isok

**Review by gaelreyrol — APPROVED** [review]

**Pull request merged**